### ISSUE TITLE
Same show in more than one storage location.

### ISSUE THREAD

**Issue by Budman1758** [Sonarr/Sonarr#1211]
Not sure if this is important to many.  I have a pretty large library of shows and as a hard drive fills up I need to move to another.  This means existing and continuing shows need to go to a new location.  What I am asking for is the ability to have more than one location for a particular show with the ability to determine where new downloads go to.  As it is now the program can only "monitor" from a single location and is unable to replace episodes with better quality or undamaged files. It also makes it too easy to mistakenly download episodes you already have.

I am not a programmer so I have no idea as to the difficulty of doing this.

My library is a home built NAS which currently has 22TB of storage.  I like to keep everything because it might be a while before I can watch it and also for for archive purposes.

Is this possible?

**Comment by markus101** [comment]
TLDR its not something we have plans to implement.

From a technical standpoint it can be solved, its not an impossible task, but it adds complications (some off the top of my head):

- Where do new files go? Last added location, location with most space, location where the rest of the season is, what if there isn't any space?
- If you deleted an old file to upgrade it, does the new file go in the same spot or in the new location? Again, what about available space issues?
- How would metadata files for applications work? Do we need to have duplicate files because the series or season is split?

When I used unRAID there were configuration options to specify where splits were allowed (which folder level), so you could force it to keep the entire series or season on one drive instead of spread out across all the drives you had., but thats a system designed to store data. I think if this is something we supported we would need something similar and people would want options, but options lead to complications, we don't want switches and knobs all over the place to control things, beyond confusing people it becomes a support nightmare.

I was tossing the idea of having support for multiple locations in a active/passive capacity, where one location would be active and any other locations would be passive and Sonarr wouldn't update files in those locations, but that only eliminates some issues and opens up a whole host of internal changes.

This is something that has been requested and something we're against implementing, not because we don't think its useful, but because its usefulness is outweighed by complications and can be solved using other systems designed for this purpose.

Drive Bender, Drive Pool and Flex Raid all have solutions that allow existing disks to be added to a pool of drives so they appear as one big drive to the OS, they can also offer data protection (duplication or RAID-5/6 like protection), SnapRAID also appears to have pooling, but not sure how it compares. I've used Drive Bender and Drive Pool in multiple setups and have had no memorable issues with either. There are also full systems such as unRAID or commercial devices such as Synology or QNAP. Even a symlink might be viable if an entire season can be contained on a single disk.

While there may be a group of users that would find this beneficial its not something that we have plans to implement.
